src/System.Runtime.Extensions/tests/System/AppDomainTests.netcoreapp.cs

```diff
@@ -16,6 +21,168 @@ public void GetSetupInformation()
             Assert.Equal(AppContext.BaseDirectory, AppDomain.CurrentDomain.SetupInformation.ApplicationBase);
             Assert.Equal(AppContext.TargetFrameworkName, AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName);
         }        
+ 
+        [Theory]
+        [MemberData(nameof(TestingCreateInstanceFromObjectHandle))]
+        static void TestingCreateInstanceFromObjectHandle(string physicalFileName, string assemblyFile, string type, string returnedFullNameType, Type exceptionType)
+        {
+            ObjectHandle oh = null;
+            object obj = null;
+
+            if (exceptionType != null)
+            {
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstanceFrom(assemblyFile: assemblyFile, typeName: type));
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstanceFromAndUnwrap(assemblyFile: assemblyFile, typeName: type));
+            }
+            else
+            {
+                oh = AppDomain.CurrentDomain.CreateInstanceFrom(assemblyFile: assemblyFile, typeName: type);
+                Assert.NotNull(oh);
+                Assert.Equal(returnedFullNameType, oh.Unwrap().GetType().FullName);
+
+                obj = AppDomain.CurrentDomain.CreateInstanceFromAndUnwrap(assemblyFile: assemblyFile, typeName: type);
+                Assert.NotNull(obj);
+                Assert.Equal(returnedFullNameType, obj.GetType().FullName);
+            }
+
+            if (exceptionType != null)
+            {
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstanceFrom(assemblyFile: assemblyFile, typeName: type, null));
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstanceFromAndUnwrap(assemblyFile: assemblyFile, typeName: type, null));
+            }
+            else
+            {
+                oh = AppDomain.CurrentDomain.CreateInstanceFrom(assemblyFile: assemblyFile, typeName: type, null);
+                Assert.NotNull(oh);
+                Assert.Equal(returnedFullNameType, oh.Unwrap().GetType().FullName);
+
+                obj = AppDomain.CurrentDomain.CreateInstanceFromAndUnwrap(assemblyFile: assemblyFile, typeName: type, null);
+                Assert.NotNull(obj);
+                Assert.Equal(returnedFullNameType, obj.GetType().FullName);
+            }
+            Assert.True(File.Exists(physicalFileName));
+        }
+
+        public static IEnumerable<object[]> TestingCreateInstanceFromObjectHandle()
+        {
+            //string physicalFileName, string assemblyFile, string typeName, returnedFullNameType, expectedException
+            yield return new object[] { "AssemblyResolveTests.dll", "AssemblyResolveTests.dll", "AssemblyResolveTests.PublicClassSample", "AssemblyResolveTests.PublicClassSample", null };
```

Comment:
can use please use ```TheoryData<U, V, W>```  property instead of ```yield return new object``` 
eg https://github.com/dotnet/corefx/blob/master/src/Common/tests/Tests/System/IO/PathInternal.Tests.cs#L68

Reply:
Same for all the other functions in this file

Reply:
There are max 5 generic args for `TheoryData`, i think i cannot use for all tests right?

Reply:
I wasnt aware of that. Good to know :)